script/lint

```diff
@@ -19,5 +19,5 @@ flake8 --doctests $files
 echo "================"
 echo "LINT with pylint"
 echo "================"
-pylint $files
+pylint $(echo "$files" | grep -v '^tests.*')
```

Comment:
Please also grep for flake8, or add a line before both where you prepare $files:

`export files =$(echo "$files" | grep -v '^tests.*')

Reply:
We do check flake8 on our tests files in CI, so we want to keep it here as well.

Reply:
I personally don't think we should change this. Although it might be some work to get this right, IMO it will lead to better tests overall. Everybody who knows this script exists should also be able to easily correct those errors.

What do you think about adding this as an optional parameter, maybe `--no-tests`?

Reply:
I personally don't think it's the right default behavior to have a file that you touched suddenly give tons of unrelated errors. It's really confusing to new contributors. We can't expect people to fix up all the existing errors in our test files.

Not opposed to adding an opt-in `--lint-tests` or something like that though.

Reply:
I do think it would be a great initiative as well though to get our tests cleaned up where we can.

Reply:
@armills did not know that. It means that `lazytox` needs some work on what is critical and what not

Reply:
I dont like adding commandline switches here... we just removed them :-)

Also tried to make as few as possible in lazytox. I only see `--` as nice there, even `--skiplint` I feel is slightly over the top

Reply:
If we don't ever test our tests for style errors, their wouldn't be any chance to improve them. Why don't we split them up and run pylint two times?

Reply:
I just ran `pylint tests/` and I see your point. Didn't thought that it would be that worst :disappointed:
If you put it in a follow up PR, add the change than (with the new logic / blacklist).

Reply:
While I'm in favor of enabling pylint for `tests/`, I think there are two problems with that:

* `pylint` is already the slowest test on travis. On my laptop running pylint over `tests/` is taking 11 minutes! And travis can often take much longer than that.
  * One solution to this would be to split up the `pylint` job into `homeassistant/` linting and `tests/` linting
* `tests/` would need custom `disable` directives. For example, `invalid-name` (too long test function names), `attribute-defined-outside-init`, `protected-access`, `len-as-condition` (magicmock's asserts can be unsafe...), `redefined-outer-name` (fixtures), `no-member` (magicmock), `no-value-for-parameter` (fixtures), etc etc
  * → We would need a separate `pylintrc` for the `tests` folder. This is of course possible.

Reply:
@OttoWinter Try executing pylint with `-j 0`
@armills This could also be usefull here

[pylint docs](https://docs.pylint.org/en/1.6.0/run.html#parallel-execution)

Reply:
@cdce8p I actually use that command for all my linting, but unfortunately it doesn't fix the speed issue *on travis*. This appeared first in #10703: travis only gives us access to one core, so `-j 0` is actually slower there.